File: RepositoryPattern/SampleApplicationForGM/network-util/src/main/java/com/example/networkutil/NetworkUtilModule.java
package com.example.networkutil;

import dagger.Module;
import dagger.Provides;

@Module
public class NetworkUtilModule {

    @Provides
    GsonUtil provideGsonUtil() {
        return GsonUtil.getInstance();
    }

    @Provides
    RetrofitNetworkUtil provideRetrofitNetworkUtil() {
        return RetrofitNetworkUtil.getInstance();
    }
}

File: RepositoryPattern/SampleApplicationForGM/network-util/src/main/java/com/example/networkutil/NetworkUtilModule.kt
package com.example.networkutil

import com.example.networkutil.RetrofitNetworkUtil.Companion.instance
import dagger.Module
import dagger.Provides

@Module
class NetworkUtilModule {
    @Provides
    fun provideGsonUtil(): GsonUtil {
        return GsonUtil.instance
    }

    @Provides
    fun provideRetrofitNetworkUtil(): RetrofitNetworkUtil {
        return instance
    }
}